File: communication_examples/subscriber.cpp
#include <stdio.h>
#include <lcm/lcm-cpp.hpp>
#include "../lcm_types/cpp/example_t.hpp"

class Handler 
{
    public:
        ~Handler() {}

        void handleMessage(const lcm::ReceiveBuffer* rbuf,
                const std::string& chan, 
                const example_t* msg)
        {
            int i;
            printf("Received message on channel \"%s\":\n", chan.c_str());
            printf("counter: %d", msg->counter);
        }
};

int main(int argc, char** argv)
{
    lcm::LCM lcm;
    if(!lcm.good())
        return 1;

    Handler handlerObject;
    lcm.subscribe("EXAMPLE", &Handler::handleMessage, &handlerObject);

    while(0 == lcm.handle());

    return 0;
}

File: communication_examples/subscriber.py
import sys

import lcm

sys.path.insert(1, '../lcm_types')
sys.path.insert(2, '../lcm_types/python')

from lcm_types.python import example_t

def my_handler(channel, data):
    msg = example_t.decode(data)
    print("   counter     = %s" % str(msg.counter))
    print("")

t = "udpm://224.0.55.55:5001?ttl=255"
lc = lcm.LCM(t)
subscription = lc.subscribe("EXAMPLE", my_handler)

try:
    while True:
        lc.handle()
except KeyboardInterrupt:
    pass
